libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/KotlinGradlePluginIT.kt

```diff
@@ -1053,4 +1053,39 @@ class KotlinGradleIT : BaseGradleIT() {
             assertTasksExecuted(":lib1:compileDebugKotlin")
         }
     }
+
+    /** Regression test for KT-38692. */
+    @Test
+    fun testIncrementalWhenNoKotlinSources() = with(
```

Comment:
nit: "test clean-up outputs on non-incremental run (when no Kotlin sources)" is more correct (like the commit message)?

Reply:
It is quite verbose IMHO, test should be easy enough to read and link to the issue provides additional context.